ci-set-env

```diff
@@ -43,6 +43,16 @@ if [ -z "$compiler" ]; then
  compiler=gcc
 fi
 $compiler --version || true
+
+case "$compiler" in
+*gcc)
+echo === implicit options/values with gcc ===
+ { $compiler -Q --help=common || true; } | grep -e '-[Wf]'
```

Comment:
just a minor note for me. Not all systems are running bash, I can't honestly remember if this syntax { } is bash specific or portable. Do  you mind to double check?

Reply:
Good point, I figured out portability is important, and double checking
this, `dash`, `ksh`, all these at least semi-POSIX shells are happy,
see also
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_09_04

`csh` is not (explicitly non-POSIX), but then, it is hopefully _not_
deployed as the script-running shell anywhere.